### PR TITLE
chore(): create a docker images for utilitis

### PR DESCRIPTION
**Pull request by DvirYo-starkware**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)

### CODE REVIEW

**Review by DvirYo-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @dan-starkware and @eranreshef-starkware)

___
*[`papyrus_utilities.Dockerfile` line 1 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-Nsi0kyv72gJEFjYfpR2:-Nsi0kyv72gJEFjYfpR3:b-b7ftdw) ([raw file](https://github.com/starkware-libs/papyrus/blob/ed872f991e0aab2d925171199d28929b0e443408/papyrus_utilities.Dockerfile#L1)):*
> ```
> # syntax = devthefuture/dockerfile-x
> ```

This line must be the first in the file.

**Review by eranreshef-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @dan-starkware and @DvirYo-starkware)

___
*[`Dockerfile` line 64 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXE-g-3DcZE3it9eR:-NtBXE-g-3DcZE3it9eS:bnp8p9x) ([raw file](https://github.com/starkware-libs/papyrus/blob/ed872f991e0aab2d925171199d28929b0e443408/Dockerfile#L64)):*
> ```Dockerfile
> 
> # Starting a new stage so that the final image will contain only the executable.
> FROM alpine:3.17.0 AS papyrus_node
> ```

why is this label needed?
___
*[`papyrus_utilities.Dockerfile` line 8 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXx44AmcCS3O2Ftnp:-NtBXx44AmcCS3O2Ftnq:b-hzwevj) ([raw file](https://github.com/starkware-libs/papyrus/blob/ed872f991e0aab2d925171199d28929b0e443408/papyrus_utilities.Dockerfile#L8)):*
> ```
> # DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 docker build -f papyrus_utilities.Dockerfile .
> 
> # TODO(dvir): consider adding the build of papyrus_utilities to the CI or to the nightly-test.
> ```

not sure we want to have a TODO in a public repo :man_shrugging: 
___
*[`papyrus_utilities.Dockerfile` line 24 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXHol0kt_Fb63L3_6:-NtBXHol0kt_Fb63L3_7:bnp8p9x) ([raw file](https://github.com/starkware-libs/papyrus/blob/ed872f991e0aab2d925171199d28929b0e443408/papyrus_utilities.Dockerfile#L24)):*
> ```
> 
> # Starting a new stage so that the final image will contain only the executables.
> FROM alpine:3.17.0 AS papyrus_utilities
> ```

why is this label needed?
___
*[`papyrus_utilities.Dockerfile` line 31 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXSmY9HLRskitjCNA:-NtBXSmY9HLRskitjCNB:b5pbj7v) ([raw file](https://github.com/starkware-libs/papyrus/blob/ed872f991e0aab2d925171199d28929b0e443408/papyrus_utilities.Dockerfile#L31)):*
> ```
> 
> # Copy the dump_declared_classes executable.
> COPY --from=utilities_builder /app/target/x86_64-unknown-linux-musl/release/dump_declared_classes /app/target/release/dump_declared_classes
> ```

this Dockerfile is missing an ENTRYPOINT directive. It is not mandatory but since I assume this container will contain multiple executables, it is recommended to have some sort of `--help` menu that can detail which executables are available.

**Review by DvirYo-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @dan-starkware and @eranreshef-starkware)

___
*[`Dockerfile` line 64 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXE-g-3DcZE3it9eR:-NtBi2Vc3ymeQmgUDyPY:b6bom62) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/Dockerfile#L64)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

why is this label needed?
</blockquote></details>

This label is not in use. I added a label for each stage to have the same structure for all the stages and make it as clear as possible what each stage does. I don't think there is any disadvantage to the labels.
___
*[`papyrus_utilities.Dockerfile` line 8 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXx44AmcCS3O2Ftnp:-NtBhu7c7zKPAoCFy6gH:bxw5tob) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/papyrus_utilities.Dockerfile#L8)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

not sure we want to have a TODO in a public repo :man_shrugging:
</blockquote></details>

We already have 360 todos in the repo.
___
*[`papyrus_utilities.Dockerfile` line 24 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXHol0kt_Fb63L3_6:-NtBipIU9zihcSK126a7:b-qzyi8y) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/papyrus_utilities.Dockerfile#L24)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

why is this label needed?
</blockquote></details>

See my first answer.
___
*[`papyrus_utilities.Dockerfile` line 31 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXSmY9HLRskitjCNA:-NtBlqHnAu6v1WU8ZEnZ:bb6un1j) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/papyrus_utilities.Dockerfile#L31)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

this Dockerfile is missing an ENTRYPOINT directive. It is not mandatory but since I assume this container will contain multiple executables, it is recommended to have some sort of `--help` menu that can detail which executables are available.
</blockquote></details>

Done

**Comment by eranreshef-starkware** [comment]
*[`papyrus_utilities.Dockerfile` line 8 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXx44AmcCS3O2Ftnp:-NtFM-un8f_fdTARIpDx:b-1z398h) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/papyrus_utilities.Dockerfile#L8)):*
<details><summary><i>Previously, DvirYo-starkware wrote…</i></summary><blockquote>

We already have 360 todos in the repo.
</blockquote></details>

:man_facepalming:

**Comment by eranreshef-starkware** [comment]
*[`Dockerfile` line 64 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtBXE-g-3DcZE3it9eR:-NtFMGac7aRwUBo1Bq_4:b-ewk48i) ([raw file](https://github.com/starkware-libs/papyrus/blob/b5c26df961af677bd7328a5d4c98f25ff568339e/Dockerfile#L64)):*
<details><summary><i>Previously, DvirYo-starkware wrote…</i></summary><blockquote>

This label is not in use. I added a label for each stage to have the same structure for all the stages and make it as clear as possible what each stage does. I don't think there is any disadvantage to the labels.
</blockquote></details>

so basically you use labels instead of comments?

**Review by eranreshef-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @dan-starkware and @DvirYo-starkware)

___
*[`papyrus_utilities.Dockerfile` line 38 at r4](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtFMpoO41i80PSAZCH9:-NtFMpoO41i80PSAZCHA:b-gk16u8) ([raw file](https://github.com/starkware-libs/papyrus/blob/f2e94fe5b25f476feef0ed16c5e7c41427a096f4/papyrus_utilities.Dockerfile#L38)):*
> ```
>  - papyrus_load_test in path /app/target/release/papyrus_load_test\n\
>  - dump_declared_classes in path /app/target/release/dump_declared_classes\n\
> Example: docker run --entrypoint /app/target/release/papyrus_load_test <image>"
> ```

1. There are multiple ways to run containers, each container runtime has different syntax
2. It is preferred to be able to run executable without supplying the full path in the run command. That can be achieved by adding the `/app/target/release` dir to the PATH
3. Please add description on what each utility does. If the utility itself has a `--help` command, it can be used here.
"Papyrus utilities container. Available utilities:\n\
- papyrus_load_test - <describe what this utility does>\n\
- dump_declared_classes - <describe what this utility does>\n\"



_Code quote:_
```
"There is no default executable for this image. Run the executables directly using the --entrypoint flag.\n\
The available executables are:\n\
 - papyrus_load_test in path /app/target/release/papyrus_load_test\n\
 - dump_declared_classes in path /app/target/release/dump_declared_classes\n\
Example: docker run --entrypoint /app/target/release/papyrus_load_test <image>"
```

**Comment by eranreshef-starkware** [comment]
*[`papyrus_utilities.Dockerfile` line 38 at r4](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtFMpoO41i80PSAZCH9:-NtQl9Z1DPOhDpmWsIj_:b-pch75c) ([raw file](https://github.com/starkware-libs/papyrus/blob/f2e94fe5b25f476feef0ed16c5e7c41427a096f4/papyrus_utilities.Dockerfile#L38)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

1. There are multiple ways to run containers, each container runtime has different syntax
2. It is preferred to be able to run executable without supplying the full path in the run command. That can be achieved by adding the `/app/target/release` dir to the PATH
3. Please add description on what each utility does. If the utility itself has a `--help` command, it can be used here.
"Papyrus utilities container. Available utilities:\n\
- papyrus_load_test - <describe what this utility does>\n\
- dump_declared_classes - <describe what this utility does>\n\"
</blockquote></details>

why does `papyrus_load_test` has a `<image>` at the end?

**Review by DvirYo-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @dan-starkware and @eranreshef-starkware)

___
*[`papyrus_utilities.Dockerfile` line 38 at r4](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-NtFMpoO41i80PSAZCH9:-NtR49bUCzB_d8affLMG:b-x1lkh4) ([raw file](https://github.com/starkware-libs/papyrus/blob/f2e94fe5b25f476feef0ed16c5e7c41427a096f4/papyrus_utilities.Dockerfile#L38)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

why does `papyrus_load_test` has a `<image>` at the end?
</blockquote></details>

This is an example of running an executable using docker runtime. The <image> at the end of the example is a placeholder for the image identifier.

**Review by eranreshef-starkware — COMMENTED** [review]
the changes are ok but this entire content shouldn't be in this repo. internal dev tools shouldn't be in a public repo.

> Reviewed 1 of 2 files at r1, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802#-:-NtUkW4D3H1TcKbi9FbP:b2zpl6f)* status: all files reviewed, 1 unresolved discussion (waiting on @dan-starkware)

**Review by dan-starkware — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1802)* status: all files reviewed, 1 unresolved discussion (waiting on @DvirYo-starkware)
